sound/soc/sof/topology.c

```diff
@@ -1959,14 +1964,32 @@ static int sof_widget_unload(struct snd_soc_component *scomp,
 
 		break;
 	case snd_soc_dapm_pga:
-
-		/* get volume kcontrol */
-		kc = &widget->kcontrol_news[0];
-		sm = (struct soc_mixer_control *)kc->private_value;
-		scontrol = sm->dobj.private;
-
-		/* free volume table */
-		kfree(scontrol->volume_table);
+		for (i = 0; i < widget->num_kcontrols; i++) {
```

Comment:
@bardliao this needs to be done irrespective of widget type I think.
previously we only had the volume component have a mixer control. so we should just have this loop without the switch.
A widget can have only type of kcontrol today. So that can be used to check which type of kcontrol to free.

Reply:
Thanks @ranj063 It is really a good suggestion. I will update my PR.